Initial submission, doug-lab, does not meet MVP.

gsmatth opened the pull request:
I will resubmit

gsmatth committed:
added basic scaffolding

gsmatth committed:
Merge pull request #1 from gsmatth/scaffold

added basic scaffolding

gsmatth committed:
added main component

gsmatth committed:
Merge pull request #2 from gsmatth/scaffold

added main component

gsmatth committed:
added basic list service

modified:   ../component/main/index.js MainController function to invoke createList function
modified:   ../component/main/main.html template to use list service to display lists
modified:   ../entry.js to require list-service
new file:   list-service.js added create list service with createList function

gsmatth committed:
Merge pull request #3 from gsmatth/listservice

added basic list service

gsmatth committed:
added delete, put, update functions on listService

modified:   ../component/main/index.js
modified:   ../entry.js
modified:   ../html/index.html
modified:   list-service.js

gsmatth committed:
Merge pull request #4 from gsmatth/listservice

added delete, put, update functions on listService

gsmatth committed:
added lists property/array to listService

*added lists array to listService so it is returned as part of object.
*modified createList, fetchList, and update list so that the local lists array is updated in addition to the backend db being updated when we modify the db.
modified:   ../component/main/index.js
modified:   list-service.js

gsmatth committed:
Merge pull request #5 from gsmatth/listservice

added lists property/array to listService

gsmatth committed:
added create-list-form components

*added createListForm directive and CreateListFormController function
new file:   ../component/create-list-form/create-list-form.html
new file:   ../component/create-list-form/create-list-form.scss
new file:   ../component/create-list-form/index.js
modified:   ../entry.js

gsmatth committed:
Merge pull request #6 from gsmatth/listservice

added create-list-form components

gsmatth committed:
create-list-form functional now

gsmatth committed:
modified maincontroller to add this.lists property

gsmatth committed:
added additional comments got list form

modified:   app/component/create-list-form/index.js
modified:   app/component/main/index.js
modified:   app/service/list-service.js

gsmatth committed:
Merge pull request #7 from gsmatth/listform

Listform

gsmatth committed:
added list component functionality

modified:   create-list-form/index.js
new file:   list/index.js
new file:   list/list.html
new file:   list/list.scss
modified:   main/index.js
modified:   main/main.html
modified:   ../entry.js
modified:   ../html/index.html
modified:   ../service/list-service.js

gsmatth committed:
Merge pull request #8 from gsmatth/listcomponent

added list component functionality

gsmatth committed:
minor debug of list components

gsmatth committed:
Merge pull request #9 from gsmatth/listcomponent

minor debug of list components

gsmatth committed:
added basic note-service.js

gsmatth committed:
added create note form functionality

new file:   app/component/create-note-form/create-note-form.html
new file:   app/component/create-note-form/create-note-form.scss
new file:   app/component/create-note-form/index.js
modified:   app/component/list/index.js
modified:   app/component/list/list.html
modified:   app/entry.js
modified:   app/service/note-service.js

gsmatth committed:
Merge pull request #10 from gsmatth/note

Note

gsmatth committed:
added note creation functionality

modified:   app/component/list/list.html
new file:   app/component/note-item/index.js
new file:   app/component/note-item/note-item.html
new file:   app/component/note-item/note-item.scss
modified:   app/entry.js
modified:   app/service/note-service.js

gsmatth committed:
added delete note functionality

modified:   app/component/list/index.js
modified:   app/component/list/list.html
modified:   app/component/list/list.scss
modified:   app/component/note-item/index.js
modified:   app/component/note-item/note-item.html
modified:   app/service/note-service.js

gsmatth committed:
Merge pull request #11 from gsmatth/note

Note

gsmatth committed:
updated package.json for testing and added first test

*added test for list service, testing the createList function.

gsmatth committed:
Merge pull request #12 from gsmatth/test

updated package.json for testing and added first test

gsmatth committed:
added tests for services

*added post, put, delete, edit tests for list service
*added post and delete tests for note services
modified:   app/entry.js
modified:   app/service/list-service.js
modified:   test/list-service-test.js
new file:   test/note-service-test.js
modified:   webpack.config.js

gsmatth committed:
Merge pull request #13 from gsmatth/test

added tests for services

fraziermork commented on lab-doug/.gitignore:
```diff
@@ -0,0 +1,83 @@
+# Created by https://www.gitignore.io/api/osx,linux,node
```
Need to ignore your build folder.

fraziermork commented on lab-doug/app/component/create-list-form/create-list-form.html:
```diff
@@ -0,0 +1,7 @@
+<form class="form-inline create-list-form" novalidate ng-submit="createListFormCtrl.createList()">
+  <div class="form-group">
+    <label for="name">First Name:</label>
```
The for attribute of a label tag needs to be the id of the element it is labelling, not the name of it.

fraziermork commented on lab-doug/app/component/create-list-form/index.js:
```diff
@@ -0,0 +1,32 @@
+    controller: ['$log', 'listService', CreateListFormController],
+    controllerAs: 'createListFormCtrl',
+    bindToController: true,
+    scope: {}
```
Why are you giving this an isolate scope?

fraziermork commented on lab-doug/app/component/create-list-form/index.js:
```diff
@@ -0,0 +1,32 @@
+    .then(() => {
+      this.list = {};
+    })
+    .catch(() => {
```
This doesn't seem great for debugging -- how do you know whether you've had an error or not?

fraziermork commented on lab-doug/app/component/list/index.js:
```diff
@@ -0,0 +1,58 @@
+    controllerAs: 'listCtrl',
+    bindToController: true,
+    scope: {
+      list: '='
```
This is a great way to do it!

fraziermork commented on lab-doug/app/component/main/index.js:
```diff
@@ -0,0 +1,41 @@
+    controller: ['listService', MainController],
+    controllerAs: 'mainCtrl',
+    bindToController: true,
+    scope: {}
```
Why the isolate scope?

fraziermork commented on lab-doug/app/component/note-item/index.js:
```diff
@@ -0,0 +1,24 @@
+    restrict: 'E',
+    replace: true,
+    template: require('./note-item.html'),
+    controller: ['$log', 'noteService', NoteItemController],
```
I would avoid defining controllers on the directive. Define the controller on the module, then specify it here as a string. If you do it that way, it's testable and can be injected w/ angular mocks. If you do it this way, you can't do that.

fraziermork commented on lab-doug/app/entry.js:
```diff
@@ -0,0 +1,27 @@
+const angular = require('angular');
+
+// angular modules
+angular.module('widgetApp', [])
```
Nice

gsmatth committed:
added changes from lab 31 grading

*debugged service tests
*added initial list controller tests
new file:   test/list-controller-test.js

gsmatth committed:
Merge pull request #14 from gsmatth/test

added changes from lab 31 grading

gsmatth committed:
added controller tests

*refactored list-service and note-service to support better testing
modified:   app/component/list/index.js
modified:   app/service/list-service.js
modified:   app/service/note-service.js
modified:   test/list-controller-test.js
modified:   test/list-service-test.js
modified:   test/note-service-test.js

gsmatth committed:
Merge pull request #15 from gsmatth/test

added controller tests

gsmatth committed:
added app-list-directive-test.js

gsmatth committed:
Merge pull request #16 from gsmatth/test

added app-list-directive-test.js

gsmatth committed:
debugged app-list-directive-test

modified:   app/component/create-note-form/create-note-form.html
modified:   test/app-list-directive-test.js

gsmatth committed:
Merge pull request #17 from gsmatth/test

debugged app-list-directive-test

gsmatth commented:
added updates to meet lab 33-34 mvp

fraziermork commented on lab-doug/app/component/list/index.js:
```diff
@@ -0,0 +1,65 @@
+    $log.debug('entered createNote() in listController() in list/index.js');
+    data.listId = this.list._id;
+    return noteService.createNote(data)
+    .then ((res) => {
```
noteService.createNote only resolves w/ res.data, so it's a little misleading to call it res at this point.

fraziermork commented on lab-doug/app/component/main/index.js:
```diff
@@ -0,0 +1,41 @@
+ * @return {array} creates the "lists" property on the MainController which references the "lists" array in listService object. If the promise "resolve" sucessfully, then a populated "lists" array is returned.   The main.html template uses angular data binding to this MainController.lists property, to populate current lists on the page.
+ */
+function MainController(listService){
+
```
If you're going to attach properties to MainController later, I would include them at the beginning as well for clarity and to avoid bugs. For example, I would have this.lists = [] at the beginning and then reset it's value to the list you get from the server when you get the server's response.
